packages/core-contracts/src/contracts/arks/MidasArk.sol

```diff
@@ -0,0 +1,342 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {IERC20Metadata} from "@openzeppelin/contracts/interfaces/IERC20Metadata.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Constants} from "@summerfi/constants/Constants.sol";
+
+import {ArkWithWithdrawalRequest, ArkParams, IArk, Ark, IArkWithWithdrawalRequest} from "../ArkWithWithdrawalRequest.sol";
+import {IDepositVault} from "../../interfaces/midas/IDepositVault.sol";
+import {IRedemptionVault, Request, RequestStatus} from "../../interfaces/midas/IRedemptionVault.sol";
+import {IMidasOracle} from "../../interfaces/midas/IMidasOracle.sol";
+import {IMToken} from "../../interfaces/midas/IMToken.sol";
+import {TokenConfig} from "../../interfaces/midas/IManageableVault.sol";
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+error InvalidIssuanceVault();
```

Comment:
Chain Security audit pointed out that errors should be defined in a separate file. It is also my preference. What do you think about it?